### DISCUSSION TITLE
N64 Controller test- Dead zone of control stick

### DISCUSSION THREAD

**Discussion by kev4cards** [sanni/cartreader#231] [Need help]
Sanni,

Would you know the dead zone of the control stick you used for your controller test ROM? I am working with the author of mupen64plus-fz to find out what the actual dead zone of the N64 control stick is. Do you happen to know the arc length of the gears, the number of notches for each gear, the arc length from one notch to the next, and the arc length of the notch-free (dead zone) section? I think if we know the number of notches that could fit in the notch-free section it would be possible create a dead zone value to use in an emulator. The range from -84 to 84 is already implemented as well as the octagon shape. Just need to find the right dead zone. Sorry for posting this in the wrong repository.

Thanks.

**Comment by sanni** [comment]
The creator of the Steel Sticks 64 has measured a new original Stick, probably using very precise measurement tools, like so:

![BrandNewOEMNeverUsed_CalibrationReference_Angles_Notes_v01 000](https://user-images.githubusercontent.com/19169494/130316419-34915aa8-3b70-4431-a283-b034f0f41e94.png)

I also did a crude measurement using pictures as reference which I compared in Photoshop and I got 3.46° joystick movement between -1 and 1. And 64.6° between -83 and 82. 
With the stick radius of 18mm that makes 1mm of play between -1 and 1, so pretty close to the dead zone from above measurements.

Stick pressed to the right until it displayed 1:
![r](https://user-images.githubusercontent.com/19169494/130317317-c0301081-1883-42cd-970d-24f11e3ac39b.jpg)

Stick pressed to the left until it displayed -1:
![l](https://user-images.githubusercontent.com/19169494/130317319-325d6663-3f51-454c-83d8-2123e432e8d5.jpg)

As for the notches in the gears, you best refer to the 3D model by eL_maN: https://www.thingiverse.com/thing:3053523

**Comment by kev4cards** [comment]
Thanks for the information. However, I'm not sure if those numbers can really be translated into digital form easily or not. That said, would you happen to know if the value of 1 has produced movement for you in any game? The reason I ask is that the programming manual for the N64 indicates a possibility of two square dead zones. One set by the N64 itself and another that can be applied by the game. The programming manual refers to a stop width in both the X and Y directions for the control stick. I assume this to be a dead zone created by the N64 system. The manual will take this stop width and subtract it from the upper limits of control stick's range. So, I am under the impression (along with few others on the N64 emulation discord servers) that everything starts from (0,0) once in-game. Do you have a way to confirm that? It may be something than can be done in an emulator otherwise. All of that said, the biggest mystery is knowing what the actual value of the stop width is. Ultimately, that is what is needed to be coded into the emulator. 7 is the given value by Nintendo for both the X and Y axis. They make the 7 look like it is centered around the axis. So, there is a possibility that the 7 is actually 14 ( meaning -7 to 7). Here is the manual page for reference: https://ultra64.ca/files/documentation/online-manuals/man-v5-2/allman52/pro-man/pro26/26-02.htm#05 .

On a side note, are the gray values shown on the steelseries testing sheet the actual values for the control stick (in a normal XY plane or is it YX?)? Also, the dead zone play values, I assume, are centered? What about the total width and total height? Knowing both of those values with the measured play would probably produce something that can be used for the emulator.

Thanks for your help.

**Comment by kev4cards** [comment]
Just an update to let you know that the author of mupen64plus-fz and I have decided to use the ratio of the two angles from your joystick movement tests as the dead zone percentage for the emulator (the other approaches I thought of earlier appear to be dependent upon the angle after further study and better understanding of the control stick's mechanics). It has already been implemented and can be found here: https://github.com/mupen64plus-ae/mupen64plus-ae/actions . This is for android only. So, if you happen to use an android device, would you mind giving some feedback on the control stick behavior when using a non-N64 controller and the touchscreen controls (if applicable)? Additionally, a comparison to a steel sticks 64 or a new-ish N64 control stick would be quite helpful. It would be greatly appreciated if you could do that. We want to replicate the feel of a new to lightly used N64 control stick as best as possible. Thanks again for all of your N64 controller work.
